src/equations/bbm_1d.jl

```diff
@@ -0,0 +1,241 @@
+@doc raw"""
+    BBMEquation1D(; bathymetry_type = bathymetry_flat,
+                  gravity_constant, D = 1.0, eta0 = 0.0, split_form = true)
+
+BBM (Benjamin–Bona–Mahony) equation in one spatial dimension. The equations are given by
+```math
+\begin{aligned}
+  \eta_t + \sqrt{gD}\eta_x + \frac{3}{2}\sqrt{\frac{g}{D}}\eta\eta_x - \frac{1}{6}D^2\eta_{xxt} &= 0.
+\end{aligned}
+```
+
+The unknown quantity of the BBM equation is the total water height ``\eta``.
+The gravitational constant is denoted by `g` and the constant bottom topography (bathymetry) ``b = \eta_0 - D``,
+where ``\eta_0`` is the constant still-water surface and ``D`` the still-water depth. The water height above
+the bathymetry is therefore given by ``h = \eta - \eta_0 + D``. The BBM equation is only implemented for ``\eta_0 = 0``.
+
+Currently, the equations only support a flat bathymetry, see [`bathymetry_flat`](@ref).
+
+The BBM equation is first described in Benjamin, Bona, and Mahony (1972).
+The semidiscretization implemented here is developed in Ranocha, Mitsotakis, and Ketcheson (2020)
+for `split_form = true` and in Linders, Ranocha, and Birken (2023) for `split_form = false`.
+If `split_form` is `true` a split form in the semidiscretization is used, which conserves
+- the total water mass (integral of ``h``) as a linear invariant
+- a quadratic invariant (integral of ``1/2\eta(\eta - 1/6D^2\eta_{xx})``), which is called here [`energy_total_modified`](@ref)
```

Comment:
Maybe mention the other form ($H^1$-like norm) for periodic boundary conditions?

Reply:
[05d3fbd](https://github.com/JoshuaLampert/DispersiveShallowWater.jl/pull/150/commits/05d3fbd740d070a5229a0f9d457bfb341353264d)